docs/concepts-messaging.md

```diff
@@ -515,16 +515,64 @@ Exclusive is the default subscription type.
 
 #### Failover
 
-In the *Failover* type, multiple consumers can attach to the same subscription. A master consumer is picked for a non-partitioned topic or each partition of a partitioned topic and receives messages. When the master consumer disconnects, all (non-acknowledged and subsequent) messages are delivered to the next consumer in line.
+In the *Failover* type, multiple consumers can attach to the same subscription. 
 
-* For partitioned topics, the broker will sort consumers by priority level and lexicographical order of consumer name. The broker will try to evenly assign partitions to consumers with the highest priority level.
-* For non-partitioned topics, the broker will pick consumers in the order they subscribe to the non-partitioned topics.
+A master consumer is picked for a non-partitioned topic or each partition of a partitioned topic and receives messages. 
 
-For example, a partitioned topic has 3 partitions, and 15 consumers. Each partition will have 1 active consumer and 4 stand-by consumers.
+When the master consumer disconnects, all (non-acknowledged and subsequent) messages are delivered to the next consumer in line.
 
-In the diagram below, **Consumer A** is the master consumer while **Consumer B** would be the next consumer in line to receive messages if **Consumer A** is disconnected.
+##### Failover | Partitioned topics
 
-![Failover subscriptions](/assets/pulsar-failover-subscriptions.svg)
+For partitioned topics, the broker sorts consumers by priority level and lexicographical order of consumer name. 
+
+The broker tries to evenly assign partitions to consumers with the highest priority level. 
+
+A consumer is selected by running a module operation `mod (partition index, consumer index)`.
+
+- If the number of partitions in a partitioned topic is **less** than the number of consumers:
+  
+  For example, in the diagram below, this partitioned topic has 2 partitions and there are 4 consumers. 
+  
+  Each partition has 1 active consumer and 1 stand-by consumer. 
+  
+    - For p0, Consumer A is the master consumer, while Consumer B would be the next consumer in line to receive messages if Consumer A is disconnected.
+
+    - For p1, Consumer C is the master consumer, while Consumer D would be the next consumer in line to receive messages if Consumer C is disconnected.
+
+  ![Failover subscriptions](/assets/pulsar-failover-subscriptions-4.svg)
+
+- If the number of partitions in a partitioned topic is **greater** than the number of consumers:
+  
+  For example, in the diagram below, this partitioned topic has 9 partitions and 3 consumers. 
```

Comment:
Out of curiosity, in this case, each partition has one consumer, so how to realize failover?